sdk/keyvault/keyvault-keys/src/cryptographyClient.ts

```diff
@@ -22,14 +12,67 @@ import {
 
 import { getTracer } from "@azure/core-tracing";
 import { Span } from "@opentelemetry/api";
+
 import { logger } from "./log";
 import { SDK_VERSION } from "./generated/utils/constants";
 import { KeyVaultClient } from "./generated/keyVaultClient";
 import { challengeBasedAuthenticationPolicy } from "../../keyvault-common/src";
-import { createHash as cryptoCreateHash, createVerify, publicEncrypt } from "crypto";
-import * as constants from "constants";
+
+import {
+  LocalSupportedAlgorithmName,
+  localSupportedAlgorithms,
+  LocalCryptographyOperationFunction,
+  isLocallySupported
+} from "./localCryptography/algorithms";
+
+import { LocalCryptographyClient } from "./localCryptographyClient";
+
+import {
+  JsonWebKey,
+  GetKeyOptions,
+  KeyVaultKey,
+  LATEST_API_VERSION,
+  CryptographyOptions,
+  CryptographyClientOptions,
+  KeyOperation
+} from "./keysModels";
+
+import {
+  EncryptionAlgorithm,
+  KeyWrapAlgorithm,
+  WrapResult,
+  UnwrapResult,
+  DecryptResult,
+  SignatureAlgorithm,
+  SignResult,
+  VerifyResult,
+  EncryptResult
+} from "./cryptographyClientModels";
+import { KeyBundle } from "./generated/models";
 import { parseKeyVaultKeysIdentifier } from "./identifier";
 
+/**
+ * Checks whether a key can be used at that specific moment,
+ * by comparing the current date with the bundle's notBefore and expires values.
+ */
+export function checkKeyValidity(keyId?: string, keyBundle?: KeyBundle) {
+  const attributes = keyBundle?.attributes || {};
+  const { notBefore, expires } = attributes;
+  const now = new Date();
+
+  if (!keyId) {
+    throw new Error("Only local cryptography operations can be performed on JsonWebKeys without kid");
```

Comment:
Why? The existing crypto client downloads a key and allows local operations if key_ops support it.

Reply:
keyId should always be populated at this point, since we do all we can to have it, as you’re describing. This error is to avoid doing the assumption that this edge case won’t happen, as in: if this ever happens, then I believe we should properly throw.

Reply:
But why can't JWKs with kids do local crypto functions? What if someone creates a local crypto client with a JWK with a kid? And the condition and phrasing just seem in opposites anyway. If a JWK doesn't have a kid, then throw an error that kids aren't allowed for local ops? I'm not understanding the logic here.